src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshalling.cs

```diff
@@ -0,0 +1,256 @@
+                    marshalMethods.AddRange(marshalType.GetMembers("OffsetOf"));
+                    marshalMethods.AddRange(marshalType.GetMembers("StructureToPtr"));
+                    marshalMethods.AddRange(marshalType.GetMembers("PtrToStructure"));
+                    _marshalMethods = marshalMethods.ToImmutable();
+                }
+            }
+
+            public void AnalyzeMethodCall(OperationAnalysisContext context)
+            {
+                IInvocationOperation invocation = (IInvocationOperation)context.Operation;
+
+                if (invocation.TargetMethod.IsGenericMethod ? _marshalMethods.Contains(invocation.TargetMethod.ConstructedFrom) : _marshalMethods.Contains(invocation.TargetMethod))
+                {
+                    bool canTransformToDisabledMarshallingEquivalent = CanTransformToDisabledMarshallingEquivalent(invocation);
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(
+                        MethodUsesRuntimeMarshallingEvenWhenMarshallingDisabled,
+                        ImmutableDictionary.Create<string, string?>().Add(CanConvertToDisabledMarshallingEquivalentKey, canTransformToDisabledMarshallingEquivalent ? "true" : null),
+                        invocation.TargetMethod.ToDisplayString()));
+                }
+            }
+
+            private static bool CanTransformToDisabledMarshallingEquivalent(IInvocationOperation invocation)
+            {
+                return invocation.TargetMethod.Name switch
+                {
+                    "OffsetOf" => false,
+                    "SizeOf" => invocation.TargetMethod.IsGenericMethod || invocation.Arguments[0].Value is ITypeOfOperation { TypeOperand.IsUnmanagedType: true },
+                    "StructureToPtr" => invocation.Arguments[0].Value.Type.IsUnmanagedType,
+                    "PtrToStructure" => invocation.Type is not null,
+                    _ => false
+                };
+            }
+
+            public void AnalyzeFunctionPointerCall(OperationAnalysisContext context)
+            {
+                var functionPointerInvocation = IFunctionPointerInvocationOperationWrapper.FromOperation(context.Operation);
+
+                if (functionPointerInvocation.GetFunctionPointerSignature().CallingConvention() == System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    return;
+                }
+
+                AnalyzeMethodSignature(context.ReportDiagnostic, functionPointerInvocation.GetFunctionPointerSignature(), ImmutableArray.Create(functionPointerInvocation.WrappedOperation.Syntax.GetLocation()));
+            }
+
+            public void AnalyzeType(SymbolAnalysisContext context)
+            {
+                INamedTypeSymbol type = (INamedTypeSymbol)context.Symbol;
+                if (type.TypeKind != TypeKind.Delegate || (_unmanagedFunctionPointerAttribute is not null && !type.HasAttribute(_unmanagedFunctionPointerAttribute)))
```

Comment:
Consider not registering the symbol action if `_unmanagedFunctionPointerAttribute` is null. This will safe few unnecessary callbacks.